docs/data/routes/docs/nextjs/troubleshooting/en.md

```diff
@@ -0,0 +1,156 @@
+
+## Errors deploying a JSS app locally
+
+```
+IMPORT ERROR(S) OCCURRED!
+Exception thrown while importing JSS app
+Exception: Sitecore.Exceptions.AccessDeniedException
+Message: AddFromTemplate - Add access required
+```
+
+This occurs when the import role doesn't have the correct permissions. See the following Knowledge Base article for the solution: https://kb.sitecore.net/articles/650791
+
+## General solution checklist
+
+- Verify that all environment variables defined in the [`.env` file](https://github.com/Sitecore/jss/blob/master/samples/nextjs/.env) file are provided. On local environments, an `.env.local` file should be for this purpose.
+
+- Check that Sitecore has an item for your API key  in the `/sitecore/system/Settings/Services/API Keys` folder and that this item's fields are populated.
+
+- Verify that the "JSS Editing Secret" is set. The`JSS_EDITING_SECRET` [value used by your Next.js app](https://github.com/Sitecore/jss/blob/cb32d3a21b87f488bd4bb5d311d556fd1f8354c4/samples/nextjs/.env#L18) should match the [value used by your Sitecore instance](https://github.com/Sitecore/jss/blob/cb32d3a21b87f488bd4bb5d311d556fd1f8354c4/samples/nextjs/sitecore/config/JssNextWeb.config#L37).
+
+- Verify that that GraphQL endpoint is correct in [`package.json`](https://github.com/Sitecore/jss/blob/cb32d3a21b87f488bd4bb5d311d556fd1f8354c4/samples/nextjs/package.json#L12) and in the Sitecore [app definition](https://github.com/Sitecore/jss/blob/cb32d3a21b87f488bd4bb5d311d556fd1f8354c4/samples/nextjs/sitecore/config/JssNextWeb.config#L72). Try opening the GraphiQL interface in a browser, using `<sitecore hostname + graphQL endpoint>/ui?sc_apikey=<api key>`, to make sure the endpoint is working
+
+- If using [ngrok](https://ngrok.com/) for any part of your local workflow, verify that you are using the `host-header` flag.
+For example,
+```
+ngrok http -host-header=rewrite 3000
+```
+
+## GraphiQL errors
+
+### Problem: GraphiQL is blank and developer tools show "Error: Mode graphql failed to advance stream"
+
+![](/assets/img/docs/nextjs/troubleshooting/graphiql-error1.png)
+
+This happens when the data in localStorage cannot be restored (For example, if you tried to open a URL with an embedded query, and the query was invalid)
+
+Solution: 
+
+First, clear all local storage from the relevant tab
+![](/assets/img/docs/nextjs/troubleshooting/graphiql-error1-solution1.png)
+
+Then, use Chrome’s task manager to kill the process before it can write to local storage again
+![](/assets/img/docs/nextjs/troubleshooting/graphiql-error1-solution2.png)
+
+After this, GraphiQL should open
+
+### Error: "XmlException: Root element is missing"
+![](/assets/img/docs/nextjs/troubleshooting/graphiql-error2.png)
+
+This error occurs when loading GraphiQL in the same browser that is logged into Sitecore.
+
+Solution: Append `sc_mode=normal` to URL
```

Comment:
Should we include the `&` or at least callout that this is a query string parm? (or a full example I suppose)